### ISSUE TITLE
Add QA for realism of quasar template

### ISSUE THREAD

**Issue by londumas** [igmhub/desi_QA#8]
Add QA for realism of quasar template

**Comment by londumas** [comment]
@alxogm, What I mean by this QA is to try to asses how the mean continuum and the variation from the mean are different or similar between data and mocks.
I would start by a simple PCA decomposition on 4 eigenvalues of the quickquasar qso and the data qso.
We currently have only asses that the mean continuum are different in the Lya forest, but we have not looked at other part of the spectrum, neither at the variations from the mean.

![difference_continuum_mock_data](https://user-images.githubusercontent.com/3166436/48216461-01106b00-e342-11e8-90ce-1ff72860b171.png)

**Comment by alxogm** [comment]
@rodrigo9303 Rodrigo now that you have more or least reproduced this plot, the difference between the data and mocks, you are close to be able to help with this QA. 

Rodrigo is trying to look at the directions given in https://github.com/imcgreer/simqso/issues/25#issuecomment-412407196

 @londumas we are a bit confused here with the PCA decomposition,  I think you shared some code with Jorge Cervantes to do this, is that right? But this should be done together with the above, or before, or alternatively?  

Sorry, if this is obvious... 

cheers 
Alma

**Comment by alxogm** [comment]
@rodrigo9303 remember also Helion's suggestion to pay attention to the Lya, CIV, CIII and MgII emission lines.

**Comment by londumas** [comment]
@alxogm and @rodrigo9303,
The way I see it there are two important things to asses before working on improvements:
- how different is the mean continuum of quasars over the all spectrum between data and mocks, this is the most important for redrock assessment.
- how different are the variations from the mean between data and mocks, this is the most important for BAO and Pk1d analysis

A good way to do this is to compute the PCA decomposition using https://github.com/sbailey/empca for 4 eighenvectors for mocks and data and look at the differences.
It will give both the mean and the variations from the mean.

We need to asses the differences before working on fixing anything. Do you agree?

I can take ~30 - 1h minutes to build a start-up desi_QA for you @rodrigo9303, but it would be better if you do it yourself with the start-up code I send you.

**Comment by alxogm** [comment]
> A good way to do this is to compute the PCA decomposition using https://github.com/sbailey/empca for 4 eighenvectors for mocks and data and look at the differences.
> It will give both the mean and the variations from the mean.
> 
> We need to asses the differences before working on fixing anything. Do you agree?
> 
@londumas  Thanks for the quick response.  I see, and yes I agree we need a more quantitative diagnostic first, before improving. 

So @rodrigo9303 could you try to start looking at the PCA? I think that will be best for you to understand what is going on, also because now we'll have a long weekend and it will be a waste of Helion's time if Helion starts this now, and we don't follow it... 

Thanks!

**Comment by londumas** [comment]
@rodrigo9303, Thanks for this.
I would use the same code for the stacking of mocks and data, using the DR14 data there on Nersc: `project/projectdirs/cosmo/data/sdss/dr14/eboss/spectro/redux/v5_10_0/`.
It would allow to have the full spectrum, which is very important for this study.
For the data, you can remove the sky lines using the mask in https://github.com/igmhub/picca/blob/master/etc/dr14-line-sky-mask.txt, and remove any `AND_MASK!=0`.
The evolution in redshift is interesting, but is quite well understood.
It is the optical depth of the Lyman-alpha in the IGM. I am not sure it is that important to look in this direction, but I might be wrong, maybe the mocks fail on some level with redshift evolution.

**Comment by londumas** [comment]
@rodrigo9303, I looked at the correlation using the Lyb forest.
As a by-product I get the continuum. You can see that it is very different between data
and mocks. I think it is really worth investigating. The plan in desi is to use both the Lya and the Lyb forests for BAO.
Thank you for the help.

![continuum](https://user-images.githubusercontent.com/3166436/50204759-2e4f3080-0366-11e9-83dc-65c63a3a4353.png)

**Comment by rodrigo9303** [comment]
@londumas, Thanks for the observation. I have updated the QA realism quasars template notebook. The previous version was wrong because I took galaxies, stars and quasars, but now it only took quasars. I didn't subtract the mean spectrum, so the first eigenvector is likely something similar to the mean when I computed the PCA decomposition. I'm currently working on fixing some parameters in SIMQSO.
